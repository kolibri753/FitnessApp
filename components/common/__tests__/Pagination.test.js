import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";
import Pagination from '../Pagination';

describe('Pagination', () => {
  it('renders correctly with default props', () => {
    const { getByText } = render(
      <Pagination
        page={1}
        totalPages={5}
        handlePrevPage={() => {}}
        handleNextPage={() => {}}
        scrollToTop={() => {}}
        setPage={() => {}}
      />
    );
    expect(getByText('<')).toBeTruthy();
    expect(getByText('1')).toBeTruthy();
    expect(getByText('>')).toBeTruthy();
  });

  it('calls handlePrevPage when prev button is pressed', () => {
    const handlePrevPageMock = jest.fn();
    const { getByText } = render(
      <Pagination
        page={2}
        totalPages={5}
        handlePrevPage={handlePrevPageMock}
        handleNextPage={() => {}}
        scrollToTop={() => {}}
        setPage={() => {}}
      />
    );
    fireEvent.press(getByText('<'));
    expect(handlePrevPageMock).toHaveBeenCalledTimes(1);
  });

  it('calls handleNextPage when next button is pressed', () => {
    const handleNextPageMock = jest.fn();
    const { getByText } = render(
      <Pagination
        page={2}
        totalPages={5}
        handlePrevPage={() => {}}
        handleNextPage={handleNextPageMock}
        scrollToTop={() => {}}
        setPage={() => {}}
      />
    );
    fireEvent.press(getByText('>'));
    expect(handleNextPageMock).toHaveBeenCalledTimes(1);
  });

  it('calls setPage with correct page number when a pagination button is pressed', () => {
    const setPageMock = jest.fn();
    const { getByText } = render(
      <Pagination
        page={2}
        totalPages={5}
        handlePrevPage={() => {}}
        handleNextPage={() => {}}
        scrollToTop={() => {}}
        setPage={setPageMock}
      />
    );
    fireEvent.press(getByText('3'));
    expect(setPageMock).toHaveBeenCalledWith(3);
  });

  it("renders correctly", () => {
		const tree = renderer.create(<Pagination />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
