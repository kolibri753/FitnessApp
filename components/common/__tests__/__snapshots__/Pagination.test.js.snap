// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#000000",
      "borderRadius": 10,
      "height": 60,
      "justifyContent": "center",
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "marginHorizontal": 5,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": undefined,
            "fontSize": 16,
          }
        }
      >
        &lt;
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "marginHorizontal": 5,
          "opacity": 0.5,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": undefined,
            "fontSize": 16,
          }
        }
      >
        &gt;
      </Text>
    </View>
  </View>
</View>
`;
